docs/build/html/_sources/methods.rst.txt

```diff
@@ -291,42 +291,56 @@ All other outputs are ``time series x XYZ`` where XYZ is the serialized list of
+current is injected into each compartment at every time step. ::
 
-  % applies 0.2 nA to each compartment
-  x.integrate(0.2)
-  % applies 0.2 nA to first compartment of three
-  x.integrate([0.2 0 0])
-  % applies a time-varying current to first compartment
-  % where S is a matrix of size time steps × compartments
-  x.integrate(S)
+  % add 0.1 nA to each compartment at each time-step
+  x.I_ext = 0.1;
 
-If ``integrate`` is passed two arguments, the first is ignored, and the second is treated as a matrix of clamped voltage. ::
+If ``I_ext`` is non-scalar, it must be a matrix of dimension ``nSteps x nComps`` where
+``nSteps`` is the number of time-steps in the simulation (i.e. ``x_t_end/x.dt``) and ``nComps`` is the number of compartments.
+Current is injected elementwise during simulation, so that the injected current at each time step for each compartment is specfied
+by an element in the matrix. ::
+
+  % add a sinusoidal current only to the first compartment
+  Ie = zeros(nSteps, nComps); Ie(:,1) = 0.1*sin(1:size(I,1));
+  x.I_ext = Ie;
+
+Clamped voltage is always added to the ``xolotl`` structure in the matrix form. ::
+
+  % voltage clamp the first compartment at -50 then 50
+  Vc = zeros(nSteps, nComps); Vc(1:5000,1) = -50; Vc(5001:10000,1) = 50;
+  x.V_clamp = Vc;
+
+.. note::
+
+  If you add new compartments to the network, remember to make sure the injected current
+  or clamped voltage matrices is the right size (i.e. ``nSteps x nComps``).
 
-  % fixes the voltage of all compartments at -50 mV
-  x.integrate([], -50])
-  % clamps the second compartment in a two-compartment network
-  x.integrate([], [NaN -50] )
 
 .. _manipulate:
 
 manipulate
 ^^^^^^^^^^
 
 Opens the GUI to permit real-time visualization of changing network parameters.
-The first argument specifies for which network properties to generate sliders (default is all of them). This takes the form of a call to the find_ function.
-The second argument specifies which visualization function should be used (default is ``plot``). ::
+An argument specifies for which network properties to generate sliders (default is all of them). This takes the form of a call to the find_ function. ::
 
   % minimum usage
   x.manipulate
   % specify sliders only for maximal conductances of HH compartment
   x.manipulate('HH*gbar')
-  % specify the myPlot function for visualization
-  x.manipulate('HH*gbar', @myPlot)
 
-.. note::
+``manipulate`` displays the results of the ``plot`` function by default.
+If the ``manipulate_plot_func`` property of ``xolotl`` is specified (e.g. ``x.manipulate_plot_func{1} = @myFunc``),
+it will display custom functions. ``manipulate_plot_func`` is a cell of function handles to all visualzation functions
+that are wrapped by ``manipulate``. ::
 
-  The functionality of the second argument has not been implemented yet.
+  % wrap two visualization functions with manipulate
+  x.manipulate_plot_func{1} = @x.plot;
+  x.manipulate_plot_func{2} = @myFIcurve;
```

Comment:
i don't think this works yet...(mulitple plot functions)...need to check

Reply:
added flag